src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IForEachLoopStatement.vb

```diff
@@ -28,7 +28,9 @@ End Class
     ]]>.Value
 
             Dim expectedOperationTree = <![CDATA[
-IForEachLoopStatement (Iteration variable: null) (LoopKind.ForEach) (OperationKind.LoopStatement) (Syntax: 'For Each s  ... Next')
+IForEachLoopStatement (LoopKind.ForEach) (OperationKind.LoopStatement) (Syntax: 'For Each s  ... Next')
+  Locals: Local_1: s As System.String
+  LoopControlVariable: ILocalReferenceExpression: s (OperationKind.LocalReferenceExpression, Type: System.String) (Syntax: 's As String')
```

Comment:
>ILocalReferenceExpression: s (OperationKind.LocalReferenceExpression, Type: System.String) (Syntax: 's As String') [](start = 23, length = 114)

This probably should be an IVariableDeclaration. Perhaps the rule should be:
- if syntax for control variable is a variable declarator, then IVariableDeclaration
- if syntax is expression that implicitly declares a local,  ILocalReferenceExpression with IsDeclaration == true
- if other expression, regular expression translation.

Reply:
Added a note: https://github.com/dotnet/roslyn/issues/22028#issuecomment-328503872